oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyStateTest.java

```diff
@@ -81,4 +100,225 @@ public void multiValuedBinarySize() throws Exception {
+
+        // Make the method accessible
+        getCompressedValueMethod.setAccessible(true);
+        byte[] result = (byte[]) getCompressedValueMethod.invoke(state);
+
+        assertNull(result);
+        assertEquals(state.getValue(Type.STRING), STRING_HUGEVALUE);
+    }
+
+    @Test
+    public void stringAboveThresholdSizeNoCompression() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
+        DocumentNodeStore store = mock(DocumentNodeStore.class);
+
+        // Get the private method
+        Method getCompressedValueMethod = DocumentPropertyState.class.getDeclaredMethod("getCompressedValue");
+
+        // Make the method accessible
+        getCompressedValueMethod.setAccessible(true);
+
+        Field compressionThreshold = DocumentPropertyState.class.getDeclaredField("DEFAULT_COMPRESSION_THRESHOLD");
+        compressionThreshold.setAccessible(true);
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(compressionThreshold, compressionThreshold.getModifiers() & ~Modifier.FINAL);
+
+        compressionThreshold.set(null, DEFAULT_COMPRESSION_THRESHOLD);
+
+        DocumentPropertyState state = new DocumentPropertyState(store, "propertyName", "\"" + STRING_HUGEVALUE + "\"", Compression.NONE);
+
+        byte[] result = (byte[]) getCompressedValueMethod.invoke(state);
+
+        assertEquals(result.length, STRING_HUGEVALUE.length() + 2 );
+
+        assertEquals(state.getValue(Type.STRING), STRING_HUGEVALUE);
+        assertEquals(STRING_HUGEVALUE, state.getValue(Type.STRING));
+
+        // Reset the value of the field
+        compressionThreshold.set(null, -1);
+    }
+
+    @Test(expected = ComparisonFailure.class)
+    public void testInterestingStrings() throws NoSuchFieldException, IllegalAccessException {
+
+        DocumentNodeStore store = mock(DocumentNodeStore.class);
+        String testString = "\"\"simple:foo\", \"cr:a\\n\\b\", \"dquote:a\\\"b\", \"bs:a\\\\b\", \"euro:a\\u201c\", \"gclef:\\uD834\\uDD1E\",\n" +
+                "                \"tab:a\\tb\", \"nul:a\\u0000b\", \"brokensurrogate:\\ud800\"";
+
+        Field compressionThreshold = DocumentPropertyState.class.getDeclaredField("DEFAULT_COMPRESSION_THRESHOLD");
+        compressionThreshold.setAccessible(true);
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(compressionThreshold, compressionThreshold.getModifiers() & ~Modifier.FINAL);
+
+        compressionThreshold.set(null, 10);
+
+        DocumentPropertyState state = new DocumentPropertyState(store, "propertyName", testString, Compression.GZIP);
+
+        String value = state.getValue(Type.STRING);
+
+        assertEquals(testString, value);
+
+    }
+
+    @Test
+    public void performanceTest() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
I fail to see the use of this versus running oak-benchmark tests with different values of `-Doak.documentMK.stringCompressionThreshold=1024` ?